src/commands/alias/set.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfCommand, parseVarArgs } from '@salesforce/sf-plugins-core';
+import { StateAggregator, Messages } from '@salesforce/core';
+import { AliasResults } from '../../types';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.load('@salesforce/plugin-settings', 'alias.set', [
+  'summary',
+  'description',
+  'examples',
+  'error.ArgumentsRequired',
+  'error.ValueRequired',
+]);
+
+export default class AliasSet extends SfCommand<AliasResults> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  // This allows varargs
+  public static readonly strict = false;
+
+  public async run(): Promise<AliasResults> {
+    const stateAggregator = await StateAggregator.getInstance();
+
+    const { args, argv } = await this.parse(AliasSet);
+
+    if (!argv.length) throw messages.createError('error.ArgumentsRequired');
+
+    const parsed = parseVarArgs(args, argv);
+
+    const results = Object.keys(parsed).map((key) => {
```

Comment:
@iowillhoit How do you feel about trying to process every entry, rather than throwing if a value is absent. Then report on success and failures, if any.

Reply:
That's probably a good call to make `config` and `alias` consistent going forward. Will update.